### ISSUE TITLE
Extra spaces in input field invalidate search Service Type field

### ISSUE THREAD

**Issue by laflannery** [department-of-veterans-affairs/va.gov-cms#12922]
Check to make sure that this is corrected for all service type fields, despite facility type selection.

## Description
When typing into the Service Type field after choosing "Community providers (in VA's network)" for Facility type, if you accidently add an extra space prior to the value, this will invalidates an otherwise valid entry and many correct suggested options will not appear in the dropdown.

For example, if I type "Dentist" into this field, there are 12 suggested matchs that appear for me to select from:
![image](https://user-images.githubusercontent.com/106678594/224359521-9d501ab1-1608-4e85-aa81-a48d9e13e039.png)

However, if I type " Dentist" (note the space before the word), only 2 suggestions appear:
![image](https://user-images.githubusercontent.com/106678594/224359670-59562dcf-1299-42a2-acc7-314529bd42e2.png)

It is not clear from my screenshots that there is even a space in the second scenario, so it would be very hard for a user to determine that they may have mistyped and that they might be missing some possible options, and in some instances (Marriage), if I add a space, then I get no options, which could be worse.

### Additional Context
There was [a previous ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/28887) on this issue in the other repo, perhaps it might help for context

## Acceptance Criteria

- [ ] All valid suggestions appear in Service Type dropdown, regardless of extra spaces

**Comment by eselkin** [comment]
Yes, this still exists
`"Dentist"` with no spaces matches 13
`"Dentist "` with 1 spaces matches 12
`"Dentist  "` with 2 spaces matches 0
Likely that the search input is not being trimmed. Also likely that search is not using Levenshtein distance so near matches are not found. If not going to use Levenshtein distance, should at least trim inputs of spaces.

**Comment by laflannery** [comment]
I want to confirm if this should be closed or not? Will another ticket be created to resolve this or is this not something that can be resolved?

**Comment by laflannery** [comment]
@jilladams and @xiongjaneg just wanted to make sure this doesn't get lost  - can you take a look at my comment above and let me know if I should reopen or make a new defect ticket?

**Comment by carlyMhayter** [comment]
BTW, development on this is effectively blocked by RI issues. Hopefully those issues are resolved by the time Sprint 13 comes around.
